integration_test/sql/migration.exs

```diff
@@ -71,6 +71,43 @@ defmodule Ecto.Integration.MigrationTest do
     end
   end
 
+  defmodule AlterColumnFromMigration do
+    use Ecto.Migration
+
+    def change do
+      create table(:modify_from_authors, primary_key: false) do
+        add :id, :integer, primary_key: true
+      end
+      create table(:modify_from_posts) do
+        add :author, references(:modify_from_authors, type: :integer)
+      end
+
+      if direction() == :up do
+        execute "INSERT INTO modify_from_authors (id) VALUES (1)"
+        execute "INSERT INTO modify_from_posts (author) VALUES (1)"
+      end
+
+      alter table(:modify_from_posts) do
+        # remove the constraints modify_from_posts_author_fkey
+        modify :author, :integer, from: references(:modify_from_authors, type: :integer)
+      end
+      alter table(:modify_from_authors) do
+        modify :id, :bigint, from: :integer
+      end
+      case PoolRepo.__adapter__ do
+        Ecto.Adapters.MySQL ->
+          execute ~s{ALTER TABLE modify_from_posts CHANGE author author_id BIGINT UNSIGNED},
+                  ~s{ALTER TABLE modify_from_posts CHANGE author_id author BIGINT UNSIGNED}
+        _ ->
+          rename table(:modify_from_posts), :author, to: :author_id
```

Comment:
Is the rename column a necessary part of this test? I would remove it from here, as it means we either cannot run this test for MySQL or we need to special case the MySQL logic, none which are desired.